x-pack/plugins/observability_ai_assistant/server/routes/conversations/route.ts

```diff
@@ -108,6 +108,61 @@ const updateConversationRoute = createObservabilityAIAssistantServerRoute({
   },
 });
 
+const updateConversationTitleBasedOnMessages = createObservabilityAIAssistantServerRoute({
+  endpoint: 'PUT /internal/observability_ai_assistant/conversation/{conversationId}/auto_title',
+  params: t.type({
+    path: t.type({
+      conversationId: t.string,
+    }),
+    body: t.type({
+      connectorId: t.string,
+    }),
+  }),
+  options: {
+    tags: ['access:ai_assistant'],
+  },
+  handler: async (resources): Promise<Conversation> => {
+    const { service, request, params } = resources;
+
+    const client = await service.getClient({ request });
+
+    if (!client) {
+      throw notImplemented();
+    }
+
+    const conversation = await client.get(params.path.conversationId);
+
+    const response = await client.chat({
+      messages: [
+        {
+          '@timestamp': new Date().toISOString(),
+          message: {
+            role: MessageRole.Assistant,
+            content: conversation.messages.slice(1).reduce((acc, curr) => {
+              return `${acc} ${curr.message.role}: ${curr.message.content}`;
+            }, 'You are a helpful assistant for Elastic Observability. Assume the following message is the start of a conversation between you and a user; give this conversation a title based on this content: '),
+          },
+        },
+      ],
+      connectorId: params.body.connectorId,
+      stream: false,
+    });
+
+    if ('object' in response && response.object === 'chat.completion') {
+      const title =
+        response.choices[0].message?.content?.slice(1, -1) ||
+        `Conversation on ${conversation['@timestamp']}`;
+
+      return client.update({
```

Comment:
this is not an atomic update though, you need to implement this in the client for that to be the case, as you need to use the `update` API, not the `index` API in Elasticsearch. if this request happens in parallel to another request that tries to update the conversation with additional messages, it will override the title we just gave it (or this one will reset some of the messages). It's sometihng we can also do later, but let's then put it no the checklist and fix it for 8.10. Basically the idea is to move this code into the client itself `client.autoTitle(conversationId)` and then use `this.dependencies.esClient.update()`.

Reply:
Will put it on the list for 8.10, want to get this stuff merged.